### ISSUE TITLE
rexml missing strscan

### ISSUE THREAD

**Issue by l0b0** [NixOS/nixpkgs#319714]
### Describe the bug

Commit 0062a1d3305118b17c3854e92db78b464313b799 seems to have broken rexml.

### Steps To Reproduce
How to reproduce the behavior:

```console
❯ nix-shell -p rubyPackages.github-pages
❯ github-pages 
/nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/resolver.rb:228:in `search_for': Unable to resolve dependency: 'rexml (>= 0)' requires 'strscan (>= 3.0.9)' (Gem::UnsatisfiableDependencyError)
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/resolver.rb:282:in `block in sort_dependencies'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/resolver.rb:276:in `each'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/resolver.rb:276:in `sort_by'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/resolver.rb:276:in `with_index'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/resolver.rb:276:in `sort_dependencies'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/vendor/molinillo/lib/molinillo/delegates/specification_provider.rb:60:in `block in sort_dependencies'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/vendor/molinillo/lib/molinillo/delegates/specification_provider.rb:77:in `with_no_such_dependency_error_handling'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/vendor/molinillo/lib/molinillo/delegates/specification_provider.rb:59:in `sort_dependencies'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/vendor/molinillo/lib/molinillo/resolution.rb:754:in `push_state_for_requirements'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/vendor/molinillo/lib/molinillo/resolution.rb:744:in `require_nested_dependencies_for'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/vendor/molinillo/lib/molinillo/resolution.rb:727:in `activate_new_spec'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/vendor/molinillo/lib/molinillo/resolution.rb:684:in `attempt_to_activate'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/vendor/molinillo/lib/molinillo/resolution.rb:254:in `process_topmost_state'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/vendor/molinillo/lib/molinillo/resolution.rb:182:in `resolve'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/vendor/molinillo/lib/molinillo/resolver.rb:43:in `resolve'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/resolver.rb:186:in `resolve'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/request_set.rb:413:in `resolve'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems/request_set.rb:425:in `resolve_current'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems.rb:224:in `finish_resolve'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems.rb:281:in `block in activate_bin_path'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems.rb:279:in `synchronize'
	from /nix/store/c4bdx4dja3h06xg1ksv3f51alp5a44li-ruby-3.1.5/lib/ruby/3.1.0/rubygems.rb:279:in `activate_bin_path'
	from /nix/store/000877rgjvbljij9wp4njjslvxbyx214-ruby3.1-github-pages-231/bin/github-pages:18:in `<main>'
```

### Expected behavior

It should work as it did in commit 291a863e866972f356967d0a270b259f46bf987f.

### Additional context

```console
git bisect start 57610d2f8f0937f39dbd72251e9614b1561942d8 2057814051972fa1453ddfb0d98badbea9b83c06
git bisect run nix-shell --packages '(import ./. {}).rubyPackages.github-pages' --pure --run 'github-pages --version'
[…]
0062a1d3305118b17c3854e92db78b464313b799 is the first bad commit
commit 0062a1d3305118b17c3854e92db78b464313b799
Author: Joé Dupuis <joe@dupuis.io>
Date:   Thu May 16 16:44:30 2024 -0700

    rexml: 3.2.6 -> 3.2.8

 pkgs/top-level/ruby-packages.nix | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)
bisect found first bad commit
```

It looks like this is related to https://github.com/replit/nixmodules/pull/350, but `nix-shell --packages '(import ./. {}).rubyPackages.github-pages' --pure --run 'github-pages --version'` currently fails on master.

### Notify maintainers

@JoeDupuis @ryantm 

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.32, NixOS, 24.11 (Vicuña), 24.11pre638038.57d6973abba7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos"`
 - channels(username): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by JoeDupuis** [comment]
Shoot! 
Looks like it's already fixed upstream: 
https://github.com/ruby/rexml/issues/132
https://github.com/ruby/rexml/issues/131
https://github.com/ruby/rexml/commit/f1df7d13b3e57a5e059273d2f0870163c08d7420#diff-75d0a867bbfd5e375164c17cf7a9df96761fb050e7308f3918ad6fe1fe12d27dL58

Looking to submit another upgrade PR. 
I just want to check if there would have been a way to catch this, so that it doesn't happen again.

**Comment by l0b0** [comment]
> I just want to check if there would have been a way to catch this, so that it doesn't happen again.

A test running `github-pages --version` would've been enough in this case. Do you want me to look into creating one, or to help you with that?

**Comment by JoeDupuis** [comment]
I confirmed that the upgrade to 3.3.0 fixes github-pages, but I wanted to make sure it doesn't break any other gems.

It looks like kramdown-rfc2629 is the only one causing issues. kramdown-rfc2629 was fixed by pinning the old version of rexml as a temporary solution for the issue. I'll pin the old version of kramdown-rfc2629 (the only difference is the rexml pin) and submit a PR upstream to remove the pin (now that rexml has been fixed).
